### ISSUE TITLE
Routable Page Mixin not Working

### ISSUE THREAD

**Issue by byfarm** [wagtail/wagtail#12479]
### Issue Summary
When I start a new wagtail project and inherit from the `RoutablePageMixin`, none of the routes work. However, when I inherit from `RoutablePage` the routing does work. This is not inline with what is in the [docs](https://docs.wagtail.org/en/v6.2/reference/contrib/routablepage.html).

### Steps to Reproduce

1. follow the [first app docs](https://docs.wagtail.org/en/stable/getting_started/tutorial.html) to start a new project, then make `HomePage` inherit from `RoutablePageMixin`.
2. Add a routable_page to `INSTALLED_APPS` and add a path with overrides.
3. Make a template to test if the context is overridden.
4. Try to visit the overridden page or extensions off of it. It should result in 404.
5. Replace `RoutablePageMixin` with `RoutablePage` and revisiting the page works.

Example code:

home/models.py
```
from django.db import models

from wagtail.models import Page
from wagtail.fields import RichTextField
from wagtail.admin.panels import FieldPanel
from wagtail.contrib.routable_page.models import RoutablePageMixin, path


class HomePage(RoutablePageMixin):
    body = RichTextField(blank=True)

    content_panels = Page.content_panels + [
        FieldPanel("body"),
    ]

    def get_context(self, request, *args, **kwargs):
        context = super().get_context(request, *args, **kwargs)
        context["thing"] = "this is default"
        return context

    @path("")
    def override_example(self, request):
        return self.render(
            request,
            context_overrides={
                "thing": "this is not default",
            },
        )

    @path("routed/")
    def routed(self, request):
        return self.render(
            request,
            context_overrides={
                "thing": "this page is extended",
            },
        )
```

templates/home/home_page.py
```
{% extends "base.html" %}

<!-- load wagtailcore_tags by adding this: -->
{% load wagtailcore_tags %}

{% block body_class %}template-homepage{% endblock %}

<!-- replace everything below with: -->
{% block content %}
<div>
    {{ page.body|richtext }}
</div>
<div>
    {{ thing }}
</div>
{% endblock %}
```
and visit localhost:8000 and /routed

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

`RoutablePageMixin` is not working as it says in the docs. I don't even see a reference to `RoutablePage` in the docs either.

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

### Technical details

-   Python version: Run `python --version`.
-   Django version: Look in your requirements.txt, or run `pip show django | grep Version`.
-   Wagtail version: Look at the bottom of the Settings menu in the Wagtail admin, or run `pip show wagtail | grep Version:`.
-   Browser version: You can use https://www.whatsmybrowser.org/ to find this out.

### Working on this

Anyone can contribute to this. View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html), add a comment to the issue once you’re ready to start.

**Comment by gasman** [comment]
Your code has `class HomePage(RoutablePageMixin):`, but the docs specify to inherit from both `RoutablePageMixin` and `Page`.

**Comment by byfarm** [comment]
Ah I see, I made a typo when I was switching between `RoutablePage` and `RoutablePageMixin`. My code really was
```
class HomePage(Page, RoutablePageMixin):
```
But I am sure you are already seeing the problem with the order of inheritance in my code. In django, the mixin needs to come before, so it should be
```
class HomePage(RoutablePageMixin, Page):
```
I tried that out and it started working! This is how it is shown in the docs as well. Learned that today! Sorry and I hope this helps someone in the future.
